### ISSUE TITLE
Improve the performance of the Kore parser

### ISSUE THREAD

**Issue by ttuegel** [kframework/kore#2189]
The Kore parser runs out of memory with the C semantics.

See also: runtimeverification/c-semantics#172

**Comment by ttuegel** [comment]
I can't profile this directly using the C semantics because crashing will prevent the profile from being written, but I can start to profile the EVM semantics, which is also rather large, and hope that this exposes the problem.

**Comment by virgil-serbanuta** [comment]
FWIW, my computer has 128GB of ram. If that's more than you used before, I can try to profile parsing the semantics if you send me some instructions about how to do it.

**Comment by ttuegel** [comment]
> I can try to profile parsing the semantics if you send me some instructions about how to do it.

We can do a quick experiment to see if 128 GB is even enough. If it's enough memory, then I can give instructions how to rebuild the backend with profiling support. If it's not enough, then you don't waste any time rebuilding.

1. Clone this pull request: runtimeverification/c-semantics#172
2. Install a frontend release. (You probably already did this.)
3. Build the semantics: in `c-semantics`, run `make BACKEND=haskell build`.
4. Try to parse the semantics: `kore-parser --no-print-definition .build/dist/lib/kcc/haskell/c-kompiled/definition.kore +RTS -s`

The extra command-line options at the end tell the runtime system to print a summary at the end, so that we can find out how much space was required.

After I wrote all that out, I tried the `kore-parser` command just to check myself, and I guess it works now? So I don't need you to test it after all. :sweat_smile: Thanks, though!
